### ISSUE TITLE
error: 'SOCKET' does not name a type  SOCKET

### ISSUE THREAD

**Issue by liwux** [openconnect/openconnect-gui#159]
@horar 
Hi, thanks for your work.
when I build openconnect-gui in Windows 10  
 Encountered  an error:
```
[ 58%] Building CXX object src/CMakeFiles/openconnect-gui.dir/gtdb.cpp.obj
In file included from C:\dev1\openconnect-gui-1.5.0\src\dialog\mainwindow.h:25:0,
                 from C:\dev1\openconnect-gui-1.5.0\src\dialog\mainwindow.cpp:20:
C:/dev1/openconnect-gui-1.5.0/include/openconnect.h:532:1: error: 'SOCKET' does not name a type
 SOCKET
 ^
[ 61%] Building CXX object src/CMakeFiles/openconnect-gui.dir/key.cpp.obj
[ 63%] Linking CXX executable ..\..\..\bin\catch_tests.exe
C:\dev1\openconnect-gui-1.5.0\src\gtdb.cpp:31:17: warning: unused parameter 'host' [-Wunused-parameter]
     const char* host,
                 ^
C:\dev1\openconnect-gui-1.5.0\src\gtdb.cpp:32:17: warning: unused parameter 'service' [-Wunused-parameter]
     const char* service,
```
How can I solve this problem? Can you help me.

**Comment by horar** [comment]
hi, sorry - it's mentioned [here]:(https://github.com/openconnect/openconnect-gui/blob/develop/docs/known_build_issues.md)

just add this include into openconnect.h, before that SOCKET type...
#include \<winsock2.h\>

**Comment by liwux** [comment]
thanks ,I made it.

**Comment by liwux** [comment]
hi, 
@horar 
Good morning
Why I will quote this mistake after successful installation.
![4](https://cloud.githubusercontent.com/assets/16595938/22814092/a67c9eda-ef8b-11e6-911c-922efc99f028.jpg)
![3](https://cloud.githubusercontent.com/assets/16595938/22814091/a67beb2a-ef8b-11e6-896e-99019156e57c.jpg)

How can I solve this problem? Can you help me.

**Comment by horar** [comment]
hi @liwux , do you install released package or you own build?

**Comment by GANHONG1995** [comment]
你也是中国人啊？ 我有个问题请教一下。如何使用命令行指定用户密码？我每次都要手动输入。我不想用gui来连接。     我指定了密码他还会让我再输入一次好坑爹啊

**Comment by horar** [comment]
please write in english, thanks.

**Comment by horar** [comment]
missing type fixed in 5ea2857
